### DISCUSSION TITLE
Screen sharing on client and ubuntu 20.04

### DISCUSSION THREAD

**Discussion by Madtoyit** [ltsp/ltsp#277] [Q&A]
we are on ubuntu 20.04. I need to enable  the Screen Sharing on the clients (settings > sharing).
The option is not activable because "no network avaiable for sharing".
This option is important for assist user on client olso to assit from a windows or pc outside LTSP environmant. 
On the server LTSP is ok, i can enable and share screen so i can use VNC on standard port 5900, but is not we need.
Tks in advance for any suggestions.

**Comment by alkisg** [comment]
Hi Madtoyit,

The shared screen is "vino-server". This is automatically launched by network-manager on a per-connection basis. Relevant commands:

```shell
$ nmcli connection show
NAME                UUID                                  TYPE      DEVICE 
Wired connection 1  74266d44-f413-38f5-ac08-17584af78e4f  ethernet  enp0s3 
$ dconf dump /org/gnome/settings-daemon/plugins/sharing/vino-server/
[/]
enabled-connections=['74266d44-f413-38f5-ac08-17584af78e4f']
```

In LTSP, we cannot use network-manager for the client connection, as then one could just click on "disable networking" and hang the client.

So you need to google for "How to use vino-server without network-manager". This isn't an LTSP-specific question.

The answer would be "just run vino-server from /etc/xdg/autostart":

> https://mail.gnome.org/archives/desktop-devel-list/2015-July/msg00002.html
> The answer is simple.
> Start vino-server through the vino-server.desktop file, by dropping it
in /etc/xdg/autostart (or the equivalent location in the user's home
directory).

**Comment by Madtoyit** [comment]
Hi Alkisg, it work perfectly.
So for all clients , i added in Startup Applications the vino-server:  /usr/lib/vino/vino-server. I can set password for vino trought the screen share in the Gnome GUI.
Olso no problem with Epoptes, it start on connection request on the next port 5901 in automatic.

Thanks.
